SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -648,6 +649,27 @@ bool USpatialNetDriver::ClientCanSendPlayerSpawnRequests()
 void USpatialNetDriver::OnGSMQuerySuccess()
 {
 	StartupClientDebugString.Empty();
+
+	const uint64 SnapshotVersion = GlobalStateManager->GetSnapshotVersion();
+	if (SPATIAL_SNAPSHOT_VERSION != SnapshotVersion) // Are we running with the same snapshot version?
+	{
+		UE_LOG(LogSpatialOSNetDriver, Error,
+			   TEXT("Your client's snapshot version does not match your deployment's snapshot version. Client version: = '%uu', Server "
+					"version = '%uu'"),
+			   SnapshotVersion, SPATIAL_SNAPSHOT_VERSION);
+
+		if (USpatialGameInstance* GameInstance = GetGameInstance())
```

Comment:
nit - this functionality is dubbed below - worth wrapping in a function. I'm also surprised by the conditional return (in the existing code). Seems we should return regardless.